src/Sylius/Bundle/CoreBundle/Resources/config/app/state_machine/sylius_order.yml

```diff
@@ -1,8 +1,36 @@
 winzou_state_machine:
     sylius_order:
+        class: "%sylius.model.order.class%"
+        property_path: state
+        graph: sylius_order
+        state_machine_class: "%sylius.state_machine.class%"
+        states:
+            cart: ~
+            new: ~
+            cancelled: ~
+            fulfilled: ~
+        transitions:
+            create:
+                from: [cart]
+                to: new
+            cancel:
+                from: [new]
+                to: cancelled
+            fulfill:
+                from: [new]
+                to: fulfilled
         callbacks:
+            before:
+                sylius_assign_number:
+                    on: "create"
+                    do: ["@sylius.order_number_assigner", "assignNumber"]
+                    args: ["object"]
             after:
-                sylius_order_confirmation_email:
+                sylius_complete_checkout:
```

Comment:
This should be on the checkout SM and should be named something like: sylius_save_checkout_completion_date? Because that's the purpose of this callback.

Also, we could rename this service, callback is too generic and is part of our "legacy" from old state machines.